src/connector/src/sink/encoder/avro.rs

```diff
@@ -395,6 +435,8 @@ fn on_field<D: MaybeData>(data_type: &DataType, maybe: D, expected: &AvroSchema)
         _ => (expected, OptIdx::NotUnion),
     };
 
+    let inner = refs.lookup(inner);
```

Comment:
`Ref` never resolves to a `Union`, while a `Union` may contain a `Ref`. As a result we check for nullable `Union` first and then resolve `Ref`. General (instead of nullability) `Union` is not supported in sink yet.